src/shelly_hap_rgb.hpp

```diff
@@ -53,19 +53,33 @@ class RGBWLight : public Component, public mgos::hap::Service {
   Type type() const override;
   std::string name() const override;
   Status Init() override;
-  void SetOutputState(const char *source);
-  void SaveState();
-  void HSVtoRGBW(const HSV &hsv, RGBW &rgbw) const;
+
   StatusOr<std::string> GetInfo() const override;
   StatusOr<std::string> GetInfoJSON() const override;
   Status SetConfig(const std::string &config_json,
                    bool *restart_required) override;
   Status SetState(const std::string &state_json) override;
 
+  void TurnOn(const std::string &source);
+  void TurnOff(const std::string &source);
+  void Toggle(const std::string &source);
+  bool IsOn() const;
+  bool IsOff() const;
+  void SetHsv(int h, int s, int v, const std::string &source);
```

Comment:
abbreviations should be fully capitalized, `SetHSV`

Reply:
also, we have `struct HSV`, why aren't we using it here?

Reply:
The reason why it is used in that way is that the HSV struct deals with ranges between 0.0f - 1.0f for every value for internally calculations. The HSV values which will come in or out has the range 0-360, 0-100, 0-100. The indention was to prevent mixing different ranges with same variables. Maybe we should implement conversion methods to the struct to deal with different ranges with the same HSV struct?

Reply:
Whats about changing the hsv to rgbw calc to always deal with the hk ranges?

Reply:
> Whats about changing the hsv to rgbw calc to always deal with the hk ranges?

I think this would be a clean way to fix it

Reply:
I removed the HSV struct because I did not see any advantages with it anymore. The HSV base is now the cfg_ struct again as before.